### ISSUE TITLE
MD057 & MD051 (relative links & link fragments) are checking wikilinks and report false positives for them

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#235]
Previously, and as far as I can tell from the documentation, MD057 and MD051 only check mdlinks (`[]()`), not wikilinks (`[[]]`). 

Recently, those two rules started checking wikilinks as well, and in doing so report false positives (i.e, existing links get reported). However, that does not seem to be the case for all wikilinks, I couldn't really reproduce when exactly this behavior occurs.

The file, as well as the heading in the file, do exist however. Notably, the LSP does not report anything, the issue seems to be only with the rumdl CLI.

test file: 
```md
# Another heading
- some text
- [[Microsoft#Windows OS]]
```

`rumdl check` output:
```
/Users/chrisgrieser/Notes/?? General/Lorem Ipsum.md:3:3: [MD051] Link fragment 'Windows OS' not found in 'Microsoft'
/Users/chrisgrieser/Notes/?? General/Lorem Ipsum.md:3:3: [MD057] Relative link 'Microsoft' does not exist
```

**Comment by rvben** [comment]
Thanks @chrisgrieser for making me aware of this regression. Fixed and tests added. This is now fixed in the latest release.
